### PR TITLE
Add Week3 HW solutions-- Salih S.

### PR DESCRIPTION
**Pull request by salih18**
**todo.js** includes Todo class
**util.js** includes Util class which reads and writes to file
**todoActions.js** includes TodoActions class which does manipulations on todos
**route.js** includes Route class which has defined api endpoints
**index.js** includes Start class which has server features in it then fires the server

### CODE REVIEW

File: week3/homework/src/todoActions.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    catch (e) {
+      res.status(400).json({ message: 'Type (description) correctly' });
+    }
+  }
+
+  async markToDo(req, res) {
+    const index = util.todos.findIndex(todo => todo.id === req.params.id);
+    const markType = req.method === 'POST' ? true : false;
+    const msg = req.method === 'POST' ? 'completed' : 'uncompleted';
```
Comment (Meno-101) [inline week3/homework/src/todoActions.js]: what about using these both together as:
```javascript
    // responsible to handle the post and delete toggling the todo item completeness.
    const isPost = req.method === 'POST' ? true : false;
    const msg = isPost ? 'completed' : 'uncompleted';
```

with a slightly descriptive comment about why `POST` it's gonna be a lot better to know what is happening.

File: week3/homework/src/routes.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      })
+      .get('/todos/:id', todoActions.readToDo)
+      .get('/todos', todoActions.readToDos)
+      .post('/todos', todoActions.createToDo)
+      .post('/todos/:id/done', todoActions.markToDo)
+      .put('/todos/:id', todoActions.updateTodo)
+      .delete('/todos/:id', todoActions.deleteTodo)
+      .delete('/todos/:id/done', todoActions.markToDo)
+      .delete('/todos/', todoActions.clearAllTodos)
+      .use('', todoActions.notFound);
```
[the remark this answers is not in the document]
Comment (salih18) [reply week3/homework/src/routes.js]: Yes, I know that. :)

File: week3/homework/src/todoActions.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      res.status(404).json({ message: 'Todo is not found' });
+  }
+
+  async readToDos(req, res) {
+    res.status(200).json({ message: 'TODO LIST', todos: util.todos });
+  }
+
+  async createToDo(req, res) {
+    try {
+      if (req.body.description.trim()) {
```
[the remark this answers is not in the document]
Comment (salih18) [reply week3/homework/src/todoActions.js]: You are right :) I just wanted to make it shorter..

File: week3/homework/src/todoActions.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        res.status(400).json({ message: 'Description is empty!' });
+      }
+    }
+    catch (e) {
+      res.status(400).json({ message: 'Type (description) correctly' });
+    }
+  }
+
+  async markToDo(req, res) {
+    const index = util.todos.findIndex(todo => todo.id === req.params.id);
```
[the remark this answers is not in the document]
Comment (salih18) [reply week3/homework/src/todoActions.js]: Thats what I couldnt see.. Thank you so much for your informative review. I need them  :smiley:

File: week3/homework/src/todoActions.js (hunk 1 above)
Comment (salih18) [reply week3/homework/src/todoActions.js]: That is what refactoring is... Thank you :)

**Pull request closed without merging**